alacritty/src/config/monitor.rs

```diff
@@ -15,18 +14,34 @@ const DEBOUNCE_DELAY: Duration = Duration::from_millis(10);
 #[cfg(not(any(target_os = "linux", target_os = "macos", target_os = "windows")))]
 const DEBOUNCE_DELAY: Duration = Duration::from_millis(1000);
 
-pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventProxy) {
-    // Canonicalize all paths, filtering out the ones that do not exist.
-    paths = paths
-        .drain(..)
-        .filter_map(|path| match fs::canonicalize(&path) {
-            Ok(path) => Some(path),
-            Err(err) => {
-                error!("Unable to canonicalize config path {:?}: {}", path, err);
-                None
-            },
-        })
-        .collect();
+pub fn watch(paths: Vec<PathBuf>, event_proxy: EventProxy) {
+    let paths = {
+        let mut paths_tmp = Vec::new();
+        for path in paths {
+            if path.exists() {
```

Comment:
Is checking for the existence of a path really necessary? In theory files can be deleted at any time and iirc paths are only passed to the monitor when the config could be loaded. So effectively this only covers for the case that config files have been deleted in the nanoseconds between parsing and monitor creation? Correct me if I'm wrong on this.

Reply:
You're right, that's a useless check.